entity-framework/core/providers/cosmos/querying.md

```diff
@@ -0,0 +1,312 @@
+
+Unfortunately, this technique is quite inefficient and can considerably increase querying costs. Cosmos DB provides a special mechanism for paginating through the result of a query, via the use of _continuation tokens_:
+
+```csharp
+CosmosPage firstPage = await context.Sessions
+    .OrderBy(s => s.Id)
+    .ToPageAsync(pageSize: 10, continuationToken: null);
+
+string continuationToken = page.ContinuationToken;
+foreach (var session in firstPage.Values)
+{
+    // Display/send the sessions to the user
+}
+```
+
+Rather than terminating the LINQ query with `ToListAsync` or similar, we use the `ToPageAsync` method, instructing it to get at most 10 items in every page (note that there may be fewer items in the database). Since this is our first query, we'd like to get results from the beginning, and pass `null` as the continuation token. `ToPageAsync` returns a `CosmosPage`, which exposes a continuation token and the values in the page (up to 10 items). Your program will typically send those values to the client, along with the continuation token; this will allow resuming the query later and fetching more results.
+
+Let's assume the user now clicks on the "Next" button in their UI, asking for the next 10 items. You can then execute the query as follows:
+
+```csharp
+CosmosPage nextPage = await context.Sessions.OrderBy(s => s.Id).ToPageAsync(10, continuationToken);
+string continuationToken = nextPage.ContinuationToken;
+foreach (var session in nextPage.Values)
+{
+    // Display/send the sessions to the user
+}
+```
+
+We execute the same query, but this time we pass in the continuation token received from the first execution; this instructs Cosmos DB to continue the query where it left off, and fetch the next 10 items. This method of paginating is extremely efficient and cost-effective compared to using `Skip` and `Take`.
+
+To learn more about pagination in Cosmos DB, [see this page](/azure/cosmos-db/nosql/query/pagination).
+
+> [!NOTE]
+> Cosmos DB does not support backwards pagination.
+>
+> `ToPageAsync` is currently annotated as experimental, since it may be replaced with a more general EF pagination API that isn't Cosmos-specific. Although using the current API will generate a compilation warning (`EF9102`), doing so should be safe - future changes may require minor tweaks in the API shape.
+
+## `FindAsync`
+
+[`FindAsync`](xref:core/change-tracking/entity-entries#find-and-findasync) is a useful API for getting a an entity by its primary key, and avoiding a database roundtrip when the entity has already been loaded and is tracked by the context.
+
+Usually, the primary key of an entity type consists e.g. of an `Id` property. When using the EF Cosmos DB provider, the primary key also contains the partition key properties; this is the case since Cosmos DB allows different partitions to contain documents with the same JSON `id` property, and so only the combined `id` and partition key uniquely identify a single document in a container:
```

Comment:
```suggestion
When using the EF Cosmos DB provider, the primary key contains the partition key properties in addition to the property mapped to the JSON `id` property; this is the case since Cosmos DB allows different partitions to contain documents with the same JSON `id` property, and so only the combined `id` and partition key uniquely identify a single document in a container:
```